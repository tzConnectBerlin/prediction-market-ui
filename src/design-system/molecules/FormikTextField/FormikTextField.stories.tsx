import { Story, Meta } from '@storybook/react/types-6-0';
import * as Yup from 'yup';
import { FastField, Formik } from 'formik';
import { FormikTextField, FormikTextFieldProps } from './FormikTextField';

export default {
  title: 'Molecule/FormikTextField',
  component: FormikTextField,
} as Meta;

const CreateQuestionSchema = Yup.object().shape({
  question: Yup.string().min(10, 'must be at least 10 characters').required('Required'),
});

const formikProps = {
  initialValues: {
    question: '',
  },
  validationSchema: CreateQuestionSchema,
  onSubmit: () => {},
};

const Template: Story<FormikTextFieldProps> = (args) => (
  <Formik {...formikProps}>
    <FastField component={FormikTextField} {...args} />
  </Formik>
);

export const Default = Template.bind({});

Default.args = {
  id: 'question-field',
  name: 'question',
  label: 'Enter a question',
  placeholder: 'Type here',
  required: true,
  // eslint-disable-next-line no-console
  handleChange: (val) => console.log(val),
};

export const OptionalField = Template.bind({});

OptionalField.args = {
  id: 'question-field',
  name: 'question',
  label: 'Enter a question',
  placeholder: 'Type here',
};

export const WithHelpMessage = Template.bind({});

WithHelpMessage.args = {
  id: 'question-field',
  name: 'question',
  label: 'Enter a question',
  placeholder: 'Type here',
  helpMessage: 'Question should be of minimum 10 characters',
};

export const WithTooltip = Template.bind({});

WithTooltip.args = {
  id: 'question-field',
  name: 'question',
  label: 'Enter a question',
  placeholder: 'Type here',
};

export const Disabled = Template.bind({});

Disabled.args = {
  id: 'question-field',
  name: 'question',
  label: 'Enter a question',
  placeholder: 'Type here',
  disabled: true,
};
