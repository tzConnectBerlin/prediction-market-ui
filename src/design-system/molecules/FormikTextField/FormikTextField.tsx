import React from 'react';
import styled from '@emotion/styled';
import {
  FormControl,
  TextField,
  TextFieldProps,
  FormHelperText,
  useTheme,
} from '@material-ui/core';
import { FieldProps } from 'formik';
import { CustomInputLabel } from '../CustomInputLabel';
import { CustomTooltipProps } from '../../atoms/CustomTooltip/CustomTooltip';

interface InternalFieldProps extends FieldProps {
  tooltipProps?: CustomTooltipProps;
  formLabelMarginTop?: number | string;
  helpMessage?: string;
  handleChange: (
    val: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>,
  ) => void | Promise<void>;
}

export type FormikTextFieldProps = InternalFieldProps & TextFieldProps;

const StyledFormControl = styled(FormControl)`
  margin-top: 1em;
`;

export const StyledTextField = styled(TextField)`
  background-color: #f5f5f5;
  padding: 4px 1em 5px;
  & .MuiInput-root {
    ::before,
    ::after,
    :hover:not(.Mui-disabled):before {
      border-bottom: none;
      border-bottom-color: transparent;
    }
  }
`;

export const FormikTextField: React.FC<FormikTextFieldProps> = ({
  form: { touched, errors, handleBlur, handleChange: formikHandleChange },
  field: { value, name },
  handleChange,
  label,
  required,
  helpMessage,
  formLabelMarginTop,
  tooltipProps,
  ...rest
}) => {
  const helperText = touched[name] ? errors[name] : '';
  const theme = useTheme();
  return (
    <StyledFormControl>
      <CustomInputLabel
        shrink
        htmlFor={name}
        label={label}
        theme={theme}
        required={required}
        marginTop={formLabelMarginTop}
        tooltipProps={tooltipProps}
      />
      {helpMessage && <FormHelperText component="span">{helpMessage}</FormHelperText>}
      <StyledTextField
        {...rest}
        name={name}
        value={value}
        onChange={(val) => {
          formikHandleChange(val);
          handleChange && handleChange(val);
        }}
        onBlur={handleBlur}
        variant="standard"
        error={touched[name] && Boolean(errors[name])}
      />
      {helperText && <FormHelperText>{helperText}</FormHelperText>}
    </StyledFormControl>
  );
};
