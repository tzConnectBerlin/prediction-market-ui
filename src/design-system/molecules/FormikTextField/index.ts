export type { FormikTextFieldProps } from './FormikTextField';
export { FormikTextField, FormikTextField as default } from './FormikTextField';
