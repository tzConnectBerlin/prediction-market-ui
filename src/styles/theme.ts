import { createTheme, PaletteOptions } from '@material-ui/core/styles';

const lightThemePalette: PaletteOptions = {
  primary: {
    dark: 'rgba(26, 117, 255, 1)',
    main: 'rgba(1, 102, 255, 1)',
    light: 'rgba(1, 102, 255, 0.12)',
    contrastText: 'rgba(255, 255, 255, 1)',
  },
  secondary: {
    dark: 'rgba(1, 102, 255, 0.10)',
    main: 'rgba(1, 102, 255, 0.12)',
    light: 'rgba(1, 102, 255, 0.6)',
    contrastText: 'rgba(1, 102, 255, 1)',
  },
  success: {
    dark: 'rgba(212, 241, 204, 1)',
    main: 'rgba(42, 184, 0, 1)',
  },
  warning: {
    dark: 'rgba(255, 229, 212, 1)',
    main: 'rgba(253, 125, 42, 1)',
  },
  error: {
    dark: 'rgba(255, 215, 215, 1)',
    main: 'rgba(255, 57, 57, 1)',
    light: 'rgba(255, 215, 215, 0.12)',
  },
  tertiary: {
    dark: 'rgba(105, 255, 209, 1)',
    main: 'rgba(57, 255, 194, 1)',
    light: 'rgba(57, 255, 194, 0.12)',
    contrastText: 'rgba(29, 34, 39, 0.87)',
  },
  grey: {
    '900': 'rgba(29, 34, 39, 1)',
    '800': 'rgba(29, 34, 39, 0.87)',
    '700': 'rgba(29, 34, 39, 0.38)',
    '600': 'rgba(29, 34, 39, 0.09)',
    '500': 'rgba(29, 34, 39, 0.07)',
    '400': 'rgba(29, 34, 39, 0.06)',
    '300': 'rgba(29, 34, 39, 0.04)',
    '200': 'rgba(29, 34, 39, 0.03)',
    '100': 'rgba(29, 34, 39, 0.02)',
  },
  text: {
    primary: 'rgba(29, 34, 39, 0.87)',
    secondary: 'rgba(29, 34, 39, 0.65)',
    disabled: 'rgba(29, 34, 39, 0.38)',
  },
  buttonText: {
    primary: 'rgba(255, 255, 255, 0.87)',
    secondary: 'rgba(255, 255, 255, 0.65)',
    disabled: 'rgba(255, 255, 255, 0.38)',
  },
  background: {
    default: 'rgba(253, 253, 255, 1)',
    paper: 'rgba(255, 255, 255, 1)',
  },
  tonalOffset: 0.05,
  mode: 'light',
};

export const lightTheme = createTheme({
  palette: lightThemePalette,
  typography: {
    fontWeightRegular: 400,
    fontWeightBold: 700,
    h1: {
      fontWeight: 700,
      fontSize: '1.875rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
    h2: {
      fontWeight: 700,
      fontSize: '1.125rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
    h3: {
      fontWeight: 700,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h4: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h5: {
      fontWeight: 700,
      fontSize: '0.625rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h6: {
      fontWeight: 700,
      fontSize: '0.625rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    subtitle1: {
      fontWeight: 700,
      fontSize: '0.75rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    body1: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 600,
      md: 900,
      lg: 1200,
      xl: 1536,
    },
  },
  zIndex: {
    tooltip: 9999,
  },
});
