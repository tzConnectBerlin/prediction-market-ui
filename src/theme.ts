import { createTheme, PaletteOptions } from '@material-ui/core/styles';

const lightThemePalette: PaletteOptions = {
  primary: {
    dark: 'rgba(26, 117, 255, 1)',
    main: 'rgba(1, 102, 255, 1)',
    light: 'rgba(1, 102, 255, 0.12)',
  },
  secondary: {
    dark: 'rgba(1, 102, 255, 0.10)',
    main: 'rgba(1, 102, 255, 0.12)',
    contrastText: 'rgba(1, 102, 255, 1)',
  },
  success: {
    dark: 'rgba(212, 241, 204, 1)',
    main: 'rgba(42, 184, 0, 1)',
    light: 'rgba(212, 241, 204, 0.12)',
  },
  warning: {
    dark: 'rgba(255, 229, 212, 1)',
    main: 'rgba(253, 125, 42, 1)',
  },
  error: {
    dark: 'rgba(255, 215, 215, 1)',
    main: 'rgba(255, 57, 57, 1)',
    light: 'rgba(255, 215, 215, 0.12)',
  },
  tertiary: {
    dark: 'rgba(105, 255, 209, 1)',
    main: 'rgba(57, 255, 194, 1)',
    light: 'rgba(57, 255, 194, 0.12)',
    contrastText: 'rgba(29, 34, 39, 0.87)',
  },
  grey: {
    '900': 'rgba(29, 34, 39, 1)',
    '800': 'rgba(29, 34, 39, 0.87)',
    '700': 'rgba(29, 34, 39, 0.38)',
    '600': 'rgba(29, 34, 39, 0.09)',
    '500': 'rgba(29, 34, 39, 0.07)',
    '400': 'rgba(29, 34, 39, 0.04)',
    '300': 'rgba(29, 34, 39, 0.03)',
    '200': 'rgba(29, 34, 39, 0.02)',
  },
  text: {
    primary: 'rgba(29, 34, 39, 0.87)',
    secondary: 'rgba(29, 34, 39, 0.54)',
    disabled: 'rgba(29, 34, 39, 0.38)',
  },
  buttonText: {
    primary: 'rgba(255, 255, 255, 0.87)',
    secondary: 'rgba(255, 255, 255, 0.54)',
    disabled: 'rgba(255, 255, 255, 0.38)',
  },
  background: {
    default: 'rgba(253, 253, 255, 1)',
    paper: 'rgba(255, 255, 255, 1)',
  },
  tonalOffset: 0.05,
  mode: 'light',
  // action: {
  //   disabledBackground: "#0166FF",
  //   disabled: "rgba(255,255,255,0.3)",
  // },
};

export const lightTheme = createTheme({
  palette: lightThemePalette,
});

export const darkTheme = createTheme({
  palette: {
    ...lightThemePalette,
    primary: {
      dark: 'rgba(214, 231, 255, 1)',
      main: 'rgba(204, 224, 255, 1)',
      light: 'rrgba(204, 224, 255, .12)',
      contrastText: 'rgba(29, 34, 39, 0.87)',
    },
    secondary: {
      dark: 'rgba(204, 224, 255, 0.14)',
      main: 'rgba(204, 224, 255, 0.12)',
      light: 'rgba(204, 224, 255, 0.14)',
      contrastText: 'rgba(204, 224, 255, 1)',
    },
    tertiary: {
      dark: 'rgba(212, 241, 204, 1)',
      main: 'rgba(212, 241, 204, 1)',
      light: 'rgba(212, 241, 204, 0.12)',
    },
    grey: {
      '900': 'rgba(255, 255, 255, 1)',
      '800': 'rgba(255, 255, 255, 0.87)',
      '700': 'rgba(255, 255, 255, 0.38)',
      '600': 'rgba(255, 255, 255, 0.2)',
      '500': 'rgba(255, 255, 255, 0.11)',
      '400': 'rgba(255, 255, 255, 0.09)',
      '300': 'rgba(255, 255, 255, 0.03)',
      '200': 'rgba(255, 255, 255, 0.02)',
    },
    text: {
      primary: 'rgba(255, 255, 255, 0.87)',
      secondary: 'rgba(255, 255, 255, 0.54)',
      disabled: 'rgba(255, 255, 255, 0.38)',
    },
    buttonText: {
      primary: 'rgba(29, 34, 39, 0.87)',
      secondary: 'rgba(29, 34, 39, 0.54)',
      disabled: 'rgba(29, 34, 39, 0.38)',
    },
    background: {
      default: 'rgba(29, 34, 39, 1)',
      paper: 'rgba(45, 49, 54, 1)',
    },
    mode: 'dark',
  },
  zIndex: {
    tooltip: 9999,
  },
});
