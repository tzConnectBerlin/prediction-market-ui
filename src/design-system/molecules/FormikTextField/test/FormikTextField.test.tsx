import { ThemeProvider } from '@material-ui/core';
import renderer from 'react-test-renderer';
import { render, waitFor, fireEvent } from '@testing-library/react';
import * as Yup from 'yup';
import { FastField, Form, Formik } from 'formik';
import { theme } from '../../../../theme';
import { FormikTextField } from '../FormikTextField';

const CreateQuestionSchema = Yup.object().shape({
  question: Yup.string().min(10, 'must be at least 10 characters').required('Required'),
  amount: Yup.number().optional(),
});

const formikProps = {
  initialValues: {
    question: '',
  },
  validationSchema: CreateQuestionSchema,
  onSubmit: () => {},
};

const baseArgs = {
  id: 'question-field',
  name: 'question',
  label: 'Enter a question',
  placeholder: 'Type here',
  required: true,
};

const defaultArgs = {
  ...baseArgs,
  handleChange: jest.fn(),
};

const WrappedComponent: React.FC<any> = (props) => (
  <ThemeProvider theme={theme}>
    <Formik {...formikProps}>
      <Form>
        <FastField component={FormikTextField} {...props} />
        <FastField
          component={FormikTextField}
          name="amount"
          id="amount"
          label="Amount"
          placeholder="Enter Amount"
        />
      </Form>
    </Formik>
  </ThemeProvider>
);

describe('Snapshot - render FormikTextField', () => {
  it('renders correctly with default props', () => {
    const inputLabel = renderer.create(<WrappedComponent {...defaultArgs} />).toJSON();
    expect(inputLabel).toMatchSnapshot();
  });
});

describe('Element testing FormikTextField Component', () => {
  it('render correctly FormikTextField with default props', async () => {
    const { findByText } = render(<WrappedComponent {...defaultArgs} />);

    waitFor(() => {
      expect(findByText(/Enter a question/i)).toBeInTheDocument();
    });
  });

  it('render correctly FormikTextField with help message', async () => {
    const { findByText } = render(
      <WrappedComponent
        {...defaultArgs}
        helpMessage="Help message example"
        formLabelMarginTop="-5%"
      />,
    );

    waitFor(() => {
      expect(findByText(/Enter a question/i)).toBeInTheDocument();
    });
  });

  it('check handleChange gets called', async () => {
    const { findByPlaceholderText } = render(<WrappedComponent {...defaultArgs} />);
    waitFor(async () => {
      const component = await findByPlaceholderText(/Type here/i);
      fireEvent.change(component, { target: { value: 'a' } });
      expect(defaultArgs.handleChange).toBeCalled();
    });
  });

  it('check handleChange is not called when not passed', async () => {
    const { findByPlaceholderText } = render(<WrappedComponent {...baseArgs} />);
    waitFor(async () => {
      const component = await findByPlaceholderText(/Type here/i);
      fireEvent.change(component, { target: { value: 'a' } });
      expect(defaultArgs.handleChange).toBeCalledTimes(0);
    });
  });

  it('triggers error for required field', async () => {
    const { findByPlaceholderText, findByText } = render(<WrappedComponent {...defaultArgs} />);
    waitFor(async () => {
      const component = await findByPlaceholderText(/Type here/i);
      fireEvent.focus(component);
      fireEvent.focusOut(component);
      expect(findByText(/Required/i)).toBeInTheDocument();
    });
  });

  it('does not triggers error for optional field', async () => {
    const { findByPlaceholderText, findByText } = render(<WrappedComponent {...defaultArgs} />);
    waitFor(async () => {
      const component = await findByPlaceholderText(/Enter Amount/i);
      fireEvent.focus(component);
      fireEvent.focusOut(component);
      expect(findByText(/Required/i)).toBeNull();
    });
  });
});
