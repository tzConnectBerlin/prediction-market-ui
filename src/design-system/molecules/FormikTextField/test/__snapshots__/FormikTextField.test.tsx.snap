// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot - render FormikTextField renders correctly with default props 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiFormControl-marginNone css-1q4elpf-MuiFormControl-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-n94ml9-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor="question"
    >
      Enter a question
      <span
        aria-hidden={true}
        className="label-asterisk MuiFormLabel-asterisk css-125iqkp-MuiFormLabel-asterisk"
      >
        â€‰
        *
      </span>
    </label>
    <div
      className="MuiFormControl-root MuiFormControl-marginNone MuiTextField-root css-ockeh6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-wnih24-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInput-input MuiInputBase-input css-1vad28r-MuiInputBase-input"
          disabled={false}
          id="question-field"
          name="question"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Type here"
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginNone css-1q4elpf-MuiFormControl-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-n94ml9-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor="amount"
    >
      Amount
    </label>
    <div
      className="MuiFormControl-root MuiFormControl-marginNone MuiTextField-root css-ockeh6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-wnih24-MuiInputBase-root-MuiInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInput-input MuiInputBase-input css-1vad28r-MuiInputBase-input"
          disabled={false}
          id="amount"
          name="amount"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter Amount"
          required={false}
          type="text"
        />
      </div>
    </div>
    
  </div>
</form>
`;
